File: Python/Samples/Bridge/UtBridge.py
import unittest
from Printers import PrinterUsual, PrinterEmergency
from Orders import FatbookOrder, GoopleOrder
from PrintStgs import * 
from PrinterCustom import PrinterCustom


class UtBridge(unittest.TestCase):

    def test_bridge(self):
        # 列印第一家廠商:產品B的訂單
        order1 = FatbookOrder(PrinterUsual())
        order1.printOrderB()

        # 列印第二家廠商:產品A的急單
        order2 = GoopleOrder(PrinterEmergency())
        order2.printOrderA()

        # 列印第二家廠商:產品B的訂單=>但該廠商並無產品B
        order3 = GoopleOrder(PrinterUsual())
        order3.printOrderB()

        self.assertTrue(True)


    def test_bridgeNstrategy(self):
        stg = FatbookPrintStg();
        order = FatbookOrder(PrinterCustom(stg))
        order.printOrderA()
        order.printOrderB()

        self.assertTrue(True)


if __name__ == '__main__':
    unittest.main()

File: CSharp/DP.UnitTest/UtBridge.cs
using System;
using System.Diagnostics;
using DP.Domain.Samples.Strategy;
using DP.Domain.Samples.Facade;
using DP.UnitTest.Utility;
using Xunit;
using Xunit.Abstractions;
using DP.Domain.Samples.Bridge;

namespace DP.UnitTest
{
    public class UtBridge
    {
        private readonly ITestOutputHelper _output;

        public UtBridge(ITestOutputHelper output)
        {
            this._output = output;
            System.Diagnostics.Trace.Listeners.Add(new XunitTraceListener(this._output));
        }

       
        [Fact]
        public void TestBridge()
        {
            #region 列印第一家廠商:產品B的訂單
            IOrder order1 = new FatbookOrder(new PrinterUsual());
            order1.PrintOrderB();
            #endregion

            #region 列印第二家廠商:產品A的急單
            IOrder order2 = new GoopleOrder(new PrinterEmergency());
            order2.PrintOrderA();
            #endregion

            #region 列印第二家廠商:產品B的訂單=>但該廠商並無產品B
            IOrder order3 = new GoopleOrder(new PrinterUsual());
            order3.PrintOrderB();
            #endregion
        }

        [Fact]
        public void TestBridgeWithStrategy()
        {
            var stg = new FatbookPrintStg();
            IOrder order = new FatbookOrder(new PrinterCostom(stg));
            order.PrintOrderA();
            order.PrintOrderB();

        }
    }
}
